tests/UnitTests/WorkflowManager.Tests/Validators/WorkflowValidatorTests.cs

```diff
@@ -0,0 +1,364 @@
+﻿using System;
+using Monai.Deploy.WorkflowManager.Common.Interfaces;
+using Monai.Deploy.WorkflowManager.Contracts.Models;
+using Monai.Deploy.WorkflowManager.Validators;
+using Moq;
+using Xunit;
+
+namespace Monai.Deploy.WorkflowManager.Test.Validators
+{
+    public class WorkflowValidatorTests
+    {
+        public WorkflowValidator WorkflowValidator { get; set; }
+
+        private readonly Mock<IWorkflowService> _workflowService;
+
+        public WorkflowValidatorTests()
+        {
+            _workflowService = new Mock<IWorkflowService>();
+            this.WorkflowValidator = new WorkflowValidator(_workflowService.Object);
+        }
+
+        [Fact]
+        public async void ValidateWorkflow_ValidatesAWorkflow_ReturnsErrorsAndHasCorrectValidationResultsAsync()
+        {
+            var workflow = new Workflow
```

Comment:
I think its important to have the 

            for (var i = 0; i < 15; i++)
            {
            ...
            }

on the tests to ensure reset works correctly, it ensures memory is being cleaned up correctly on workflow validator reset.

Reply:
Ah I wondered why it was doing this, - I removed it as it was doing extra work without any reason to. I'll add it back with a comment to document its purpose.